src/autoscaler/scalingengine/server/server_test.go

```diff
@@ -231,5 +231,47 @@ var _ = Describe("Server", func() {
 				})
 			})
 		})
+
+		Context("when getting active schedule", func() {
+			BeforeEach(func() {
+				uPath, err := route.Get(routes.GetActiveScheduleRoute).URLPath("appid", "test-app-id")
+				Expect(err).NotTo(HaveOccurred())
+				urlPath = uPath.Path
+				bodyReader = nil
+				method = http.MethodGet
+			})
+
+			Context("when requesting correctly", func() {
+				It("should return 200", func() {
+					Expect(err).ToNot(HaveOccurred())
+					Expect(rsp.StatusCode).To(Equal(http.StatusOK))
+					rsp.Body.Close()
+				})
+			})
+
+			Context("when using the wrong method", func() {
+				BeforeEach(func() {
+					method = http.MethodDelete
+				})
+
+				It("should return 404", func() {
```

Comment:
Currently the mux library implementation cannot distinguish between 404 and 405
Looked at the code, the routes and methods don't seem to be connected

https://github.com/gorilla/mux/issues/6